flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/handlers/JarRunHandlerParameterTest.java

```diff
@@ -219,15 +225,40 @@ public void testConfigurationViaJsonRequest() throws Exception {
 	}
 
 	@Test
-	public void testParameterPrioritization() throws Exception {
+	public void testParameterPrioritizationWithProgArgsAsString() throws Exception {
+		testParameterPrioritization(PROG_ARGS, null);
+	}
+
+	@Test
+	public void testParameterPrioritizationWithProgArgsAsList() throws Exception {
+		testParameterPrioritization(null, PROG_ARGS);
+	}
+
+	@Test
+	public void testFailIfProgArgsAreAsStringAndAsList() throws Exception {
+		try {
+			testParameterPrioritization(PROG_ARGS, PROG_ARGS);
+			fail("IllegalArgumentException is excepted");
+		} catch (IllegalArgumentException e) {
+			// expected
+		}
+	}
+
+	private void testParameterPrioritization(List<String> programArgs, List<String> programArgsList) throws Exception {
+		List<String> args = programArgs == null ? programArgsList : programArgs;
```

Comment:
This code is way to confusing. It strikes me as very odd that `args` is never null, and the naming scheme is weird (args should be "argsList", "argsStr" should be "args" to be somewhat consistent with the method arguments).

I would pass in a single collection as arguments and a simple enum for how arguments should be passed.